Added transform for biodomain_info

jaclynbeck-sage opened the pull request:
This transform takes the `genes_biodomains` source file and pulls out the unique values from the `biodomain` column only. This results in a single-column DataFrame with the column `name`, with 19 biodomains. It is set up this way so there is room to add more columns/info to this new DataFrame later if we want. 

The JSON output of this single-column DataFrame looks like this:
```
[
  {
    "name": "APP Metabolism"
  },
  {
    "name": "Apoptosis"
  },
...
```

*Tests for this transform*
2 tests are run: 1 with perfect input and no missing data, and 1 with missing data from at least one column, covering all columns. Both of these should pass. 

For the perfect data scenario, the biodomains are listed out of alphabetical order and have duplicate entries. We expect the end result to be unique biodomain names, sorted. 

For the imperfect data scenario, I've also added a line for biodomain "Apoptosis" which is missing the critical "biodomain" column, and is the only entry for Apoptosis. We expect that this biodomain will be skipped due to having an NA name, AND there shouldn't be any blank entries in the output file. 

The output files for perfect and imperfect data are currently identical, but may not always be so I've made them as 2 separate files. 

There is currently no failure case for this transform.

jaclynbeck-sage committed:
Added transform for biodomain_info

jaclynbeck-sage commented:
@BWMac I want to make sure I've added the new transform correctly!

BWMac reviewed (commented):
It looks good to me!

I would recommend adding the test for the new transform in the same PR (if you weren't already planning to) as well so you don't need to come back to it.

JessterB reviewed (approved):
Looks good, thank you!

thomasyu888 commented:
@jaclynbeck-sage I second @BWMac's suggestion. It's best practice to add tests when contributing new features so we know the code is working as expected.

We may even want to make this a 'requirement' moving forward

jaclynbeck-sage committed:
Added tests for new biodomain_info transform

jaclynbeck-sage commented:
I can't believe I forgot to write a test after we've been talking about it constantly! \**facepalm*\*
Updated to add tests for the transform, and I updated the PR description to describe the tests too.

thomasyu888 commented on src/agoradatatools/etl/transform/biodomain_info.py:
```diff
@@ -0,0 +1,26 @@
+        .drop_duplicates()
+        .reset_index()
+        .drop(columns="index")
+        .rename(columns={"biodomain": "name"})
```
Nit (nitpicky): I'm ok with this here, but just wanted to hear your thoughts why we aren't using the `column_rename` or `agora_rename` functionality.

jaclynbeck-sage replied:
Oh, yeah I'll do that instead. That seems cleaner.

BWMac reviewed (approved):
Nice! :rocket:

No worries about forgetting, that's what code review is for :grin:

jaclynbeck-sage committed:
Changed biodomain_info transform to use column_rename instead of manual rename in the function

thomasyu888 reviewed (approved):
LGTM!
